cpp/test/core/container_extensibility.cpp

```diff
@@ -123,8 +123,8 @@ TEST_CASE_BEGIN(MultiIndexTest)
                 SimpleStructView, 
                 boost::multi_index::indexed_by<
                     boost::multi_index::sequenced<>,
-                    ordered_non_unique_field<SimpleStructView::Schema::var::m_str>,
-                    ordered_non_unique_field<SimpleStructView::Schema::var::m_uint64>
+                    ordered_non_unique_field<SimpleStructView::Schema::var::m_str, get_field_template<SimpleStructView::Schema::var::m_str>::type>,
+                    ordered_non_unique_field<SimpleStructView::Schema::var::m_uint64, get_field_template<SimpleStructView::Schema::var::m_uint64>::type>
```

Comment:
I simplified this, and moved all logic into ordered_non_unique_field<>, please check cpp/test/core/multi_index.h

vc12 doesn't support constexp - ordered_non_unique_field requires member field address in compile time, which can be resolved via FieldTemplate.